<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>597</width>
    <height>666</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Controller</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="7" column="0">
    <widget class="QLabel" name="status_display">
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QGroupBox" name="sequencer_group">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Sequencer</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Program</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="program_path_line_edit"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="choose_program_button">
          <property name="text">
           <string>Choose program</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="edit_program_button">
          <property name="text">
           <string>Edit program</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="run_program_analyze_button">
          <property name="text">
           <string>Run program and analyze</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="export_data_button">
          <property name="text">
           <string>To data_exported</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="control_mode_group">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Control mode</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Current mode:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="mode_status">
        <property name="text">
         <string>Manual</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="mode_button">
        <property name="text">
         <string>Auto</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="COM_group">
     <property name="title">
      <string>COM Port</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <widget class="QLineEdit" name="COM_port">
        <property name="text">
         <string>COM4</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="COM_status">
        <property name="text">
         <string>Closed</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="COM_button">
        <property name="text">
         <string>Open</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="manual_control_group">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Manual control</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="2" column="1">
       <widget class="QGroupBox" name="chamber2_group">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="title">
         <string>Chamber2</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QCheckBox" name="C2_AOM_Cool">
           <property name="text">
            <string>AOM Cool and Init</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="C2_AOM_Detect">
           <property name="text">
            <string>AOM Detect</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="C2_EOM_Cooling">
           <property name="text">
            <string>EOM Cooling</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="C2_EOM_Initialize">
           <property name="text">
            <string>EOM Initialize</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="C2_Microwave">
           <property name="text">
            <string>Microwave</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Phase:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="C2_Phase"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QGroupBox" name="chamber1_group">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="title">
         <string>Chamber1</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QCheckBox" name="C1_AOM_Cool">
           <property name="text">
            <string>AOM Cool and Init</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="C1_AOM_Detect">
           <property name="text">
            <string>AOM Detect</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="C1_EOM_Cooling">
           <property name="text">
            <string>EOM Cooling</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="C1_EOM_Initialize">
           <property name="text">
            <string>EOM Initialize</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="C1_Microwave">
           <property name="text">
            <string>Microwave</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Phase:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="C1_Phase"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="read_manaul_output_button">
        <property name="text">
         <string>Read current manual output settings</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QCheckBox" name="single_shot_checkbox">
          <property name="text">
           <string>Single shot status</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="send_pulse_button">
          <property name="text">
           <string>Send pulse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QGroupBox" name="config_file_group">
     <property name="title">
      <string>Config file</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="1" column="0">
       <widget class="QPushButton" name="reload_config_button">
        <property name="text">
         <string>Reload from config file</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="edit_config_button">
        <property name="text">
         <string>Edit config file</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="save_config_button">
        <property name="text">
         <string>Save current settings to config file</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="QLabel" name="config_file_label">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QGroupBox" name="application_group">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Application</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="1" column="2">
       <widget class="QPushButton" name="pushButton_6">
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_5">
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="pulse_rabi_button">
        <property name="text">
         <string>Rabi with pulse laser</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="repeat_run_button">
        <property name="text">
         <string>Repeat run</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="pulse_counter_button">
        <property name="text">
         <string>pulse counter</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pushButton_3">
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="pushButton_7">
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="pushButton_8">
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="pushButton_9">
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>C1_AOM_Cool</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C1_AOM_cooling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C1_AOM_Detect</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C1_AOM_detect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C1_EOM_Cooling</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C1_EOM_cooling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C1_EOM_Initialize</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C1_EOM_initialize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C1_Microwave</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C1_microwave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C1_Phase</sender>
   <signal>editingFinished()</signal>
   <receiver>Form</receiver>
   <slot>C1_phase()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C2_AOM_Cool</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C2_AOM_cooling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C2_AOM_Detect</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C2_AOM_detect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>521</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C2_EOM_Cooling</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C2_EOM_cooling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C2_EOM_Initialize</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C2_EOM_initialize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>521</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C2_Phase</sender>
   <signal>editingFinished()</signal>
   <receiver>Form</receiver>
   <slot>C2_phase()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>544</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>C2_Microwave</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>C2_microwave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>COM_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>open_COM_port()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>choose_program_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>set_program_path()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>285</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edit_program_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>edit_program()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mode_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>change_mode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>542</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>run_program_analyze_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>run_program_analyze()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>440</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_data_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>export_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reload_config_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>reload_config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_config_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>save_config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>507</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>single_shot_checkbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>single_shot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>read_manaul_output_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>read_manual_output()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>send_pulse_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>send_pulse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>469</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>274</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edit_config_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>edit_config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>repeat_run_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>repeat_run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>485</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulse_rabi_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>pulse_rabi()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>485</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>set_program_path()</slot>
  <slot>edit_program()</slot>
  <slot>C1_AOM_cooling()</slot>
  <slot>C1_AOM_detect()</slot>
  <slot>C1_EOM_cooling()</slot>
  <slot>C1_EOM_initialize()</slot>
  <slot>C1_microwave()</slot>
  <slot>C1_phase()</slot>
  <slot>C2_AOM_cooling()</slot>
  <slot>C2_AOM_detect()</slot>
  <slot>C2_EOM_cooling()</slot>
  <slot>C2_EOM_initialize()</slot>
  <slot>C2_microwave()</slot>
  <slot>C2_phase()</slot>
  <slot>open_COM_port()</slot>
  <slot>change_mode()</slot>
  <slot>run_program_analyze()</slot>
  <slot>export_data()</slot>
  <slot>reload_config()</slot>
  <slot>save_config()</slot>
  <slot>single_shot()</slot>
  <slot>read_manual_output()</slot>
  <slot>send_pulse()</slot>
  <slot>edit_config()</slot>
  <slot>repeat_run()</slot>
  <slot>pulse_rabi()</slot>
 </slots>
</ui>
